linux/install_centos7_nginx.sh

```diff
@@ -33,20 +34,24 @@ if [[ "$PGVER" =~ ^(pg94|pg95|pg96|pg10)$ ]]; then
 	bash -eux step03_all_postgres.sh
 fi
 
-cp settings.env settings-web.env step04_all_omero.sh setup_omero_db.sh ~omero
-su - omero -c "OMEROVER=$OMEROVER ICEVER=$ICEVER bash -eux step04_all_omero.sh"
+cp step01_centos7_ice_venv.sh settings.env settings-web.env step04_all_omero.sh setup_omero_db.sh ~omero
+
+# Create a virtual env to install Ice Python binding as the omero user
+su - omero -c "VIRTUALENV=$VIRTUALENV bash -eux step01_centos7_ice_venv.sh"
+
+su - omero -c "OMEROVER=$OMEROVER ICEVER=$ICEVER VIRTUALENV=$VIRTUALENV bash -eux step04_all_omero.sh"
 
 su - omero -c "bash setup_omero_db.sh"
 
-OMEROVER=$OMEROVER ICEVER=$ICEVER bash -eux step05_centos7_nginx.sh
+OMEROVER=$OMEROVER VIRTUALENV=$VIRTUALENV bash -eux step05_centos7_nginx.sh
 
 if [ "$WEBSESSION" = true ]; then
-	bash -eux step05_2_websessionconfig.sh
+	su - omero -c "VIRTUALENV=$VIRTUALENV bash -eux step05_2_websessionconfig.sh"
 fi
 
 #If you don't want to use the systemd scripts you can start OMERO manually:
-#su - omero -c "OMERO.server/bin/omero admin start"
-#su - omero -c "OMERO.server/bin/omero web start"
+#su - omero -c ". $VIRTUALENV/bin/activate;OMERODIR=/home/omero/OMERO.server omero admin start"
+#su - omero -c ". $VIRTUALENV/bin/activate;OMERODIR=/home/omero/OMERO.server omero web start"
```

Comment:
Have you looked at having OMERODIR in settings.env? Also: Simon had had some luck with just setting PATH & VIRTUAL_ENV rather than needing to activate the virtualenv if that would simplify things.